### PR TITLE
[sp] fix async print output

### PR DESCRIPTION
**Pull request by shrey-mage**
# Summary
- asynchronous print output is now combined appropriately when rendered

# Tests

cc: @tommydangerous @skunichetty @dy46 @johnson-mage

### CODE REVIEW

File: mage_ai/frontend/components/CodeBlock/CodeOutput/index.tsx [hunk 1]
```diff
@@ -79,6 +79,29 @@ function CodeOutput({
...
 
+  const combineTextData = (data) => (Array.isArray(data) ? data.join('\n') : data);
+
+  const combinedMessages = useMemo(() => messages.reduce((arr, curr) => {
+    const last = arr.at(-1);
+    
+    if (DATA_TYPE_TEXTLIKE.includes(last?.type) && last?.type === curr.type) {
+      last.data += combineTextData(curr.data);
+    }
+    else if (DATA_TYPE_TEXTLIKE.includes(curr?.type)) {
```
Comment (johnson-mage) [inline mage_ai/frontend/components/CodeBlock/CodeOutput/index.tsx:90]: move to line before

File: mage_ai/frontend/components/CodeBlock/CodeOutput/index.tsx [hunk 2]
```diff
@@ -79,6 +79,29 @@ function CodeOutput({
...
+      arr.push({
+        ...curr,
+        data: combineTextData(curr.data),
+      });
+    }
+    else {
```
Comment (johnson-mage) [inline mage_ai/frontend/components/CodeBlock/CodeOutput/index.tsx:96]: doesn't need to be on newline

**Comment by skunichetty** [comment]
Data table doesn't seem to be printed correctly: 
image

**Comment by skunichetty** [comment]
Data Table sometimes doesn't render properly: 
image

**Comment by skunichetty** [comment]
Another issue: shebangs in cell don't concatenate, report as an array rather than a continuous block of text.
image

**Review by skunichetty — APPROVED** [review]
Everything seems to look good now, at least from my testing.

**Pull request merged**